b2share/factory.py

```diff
@@ -1,192 +1,178 @@
+    folder (``<instance_folder>/templates``).
     """
-    # Create the REST API Flask application
-    api = create_api(**kwargs)
-    api.config.update(
-        APPLICATION_ROOT='/api'
+    # This is the only place customize the Flask application right after
+    # it has been created, but before all extensions etc are loaded.
+    local_templates_path = os.path.join(app.instance_path, 'theme/templates')
+    if os.path.exists(local_templates_path):
+        # Let's customize the template loader to look into packages
+        # and application templates folders.
+        app.jinja_loader = ChoiceLoader([
+            FileSystemLoader(local_templates_path),
+            app.jinja_loader,
+        ])
+
+    app.jinja_options = dict(
+        app.jinja_options,
+        cache_size=1000,
+        bytecode_cache=BytecodeCache(app)
     )
-    app_ui = Flask(__name__,
-                   static_folder=os.environ.get(
-                       'B2SHARE_UI_PATH',
-                       os.path.join(api.instance_path, 'static')),
-                   static_url_path='',
-                   instance_path=api.instance_path)
-
-    add_routes(app_ui)
-
-    api.wsgi_app = DispatcherMiddleware(app_ui.wsgi_app, {
-        '/api': api.wsgi_app
-    })
-    if api.config.get('WSGI_PROXIES'):
-        wsgi_proxies = api.config.get('WSGI_PROXIES')
-        assert(wsgi_proxies > 0)
-        api.wsgi_app = ProxyFix(api.wsgi_app,
-                                num_proxies=api.config['WSGI_PROXIES'])
-
-    check_configuration(api.config, api.logger)
-
-    return api
-
-
-def add_routes(app_ui):
-    @app_ui.route('/')
-    def root():
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/help', defaults={'path': ''})
-    @app_ui.route('/help/', defaults={'path': ''})
-    @app_ui.route('/help/<path:path>')
-    def serve_help(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/communities', defaults={'path': ''})
-    @app_ui.route('/communities/', defaults={'path': ''})
-    @app_ui.route('/communities/<path:path>')
-    def serve_communities(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/user', defaults={'path': ''})
-    @app_ui.route('/user/', defaults={'path': ''})
-    @app_ui.route('/user/<path:path>')
-    def serve_user(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/records', defaults={'path': ''})
-    @app_ui.route('/records/', defaults={'path': ''})
-    @app_ui.route('/records/<path:path>')
-    def serve_records(path):
-        return app_ui.send_static_file('index.html')
-
-    @app_ui.route('/search', defaults={'path': ''})
-    @app_ui.route('/search/', defaults={'path': ''})
-    @app_ui.route('/search/<path:path>')
-    def serve_search(path):
-        return app_ui.send_static_file('index.html')
 
+    invenio_config_loader(app, **kwargs_config)
```

Comment:
B2SHARE V2 doesn't call invenio_config_loader directly. It uses `create_conf_loader` from invenio_config.